test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs

```diff
@@ -98,21 +66,11 @@ protected virtual void AddAdditionalServices(IServiceCollection services)
             services.AddSingleton<IApplicationEnvironment>(
                 new TestApplicationEnvironment(applicationEnvironment, applicationName, applicationRoot));
 
-            var hostingEnvironment = new HostingEnvironment();
-            hostingEnvironment.Initialize(applicationRoot, new WebApplicationOptions(), configuration: null);
-            services.AddSingleton<IHostingEnvironment>(hostingEnvironment);
-
             // Inject a custom assembly provider. Overrides AddMvc() because that uses TryAdd().
             var assemblyProvider = new StaticAssemblyProvider();
             assemblyProvider.CandidateAssemblies.Add(startupAssembly);
             services.AddSingleton<IAssemblyProvider>(assemblyProvider);
-
-            AddAdditionalServices(services);
-
-            if (configureServices != null)
-            {
-                configureServices(services);
-            }
+            services.AddSingleton<ILoggerFactory>(NullLoggerFactory.Instance);
```

Comment:
This is a bug in the `TestHost` implementation. Some apps register a ConsoleLogger which results in a null ref here: https://github.com/aspnet/Hosting/blob/0f0999161d5dfb12feba90f26bbc05795fdf00b3/src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs#L144

```C#
System.NullReferenceException : Object reference not set to an instance of an object.
Stack Trace:
   at Microsoft.AspNet.Http.Internal.DefaultHttpRequest.get_Protocol()
   at Microsoft.AspNet.Hosting.Internal.HostingLoggerExtensions.HostingRequestStarting.ToString()
   at Microsoft.AspNet.Hosting.Internal.HostingLoggerExtensions.HostingRequestStarting.<>c.<.cctor>b__7_0(Object state, Exception exception)
   at Microsoft.Extensions.Logging.Console.ConsoleLogger.Log(LogLevel logLevel, Int32 eventId, Object state, Exception exception, Func`3 formatter)
   at Microsoft.Extensions.Logging.Logger.Log(LogLevel logLevel, Int32 eventId, Object state, Exception exception, Func`3 formatter)
```
I'll file a bug for it.